datafusion/core/src/execution/context.rs

```diff
@@ -495,6 +495,7 @@ impl SessionContext {
         }
 
         let input = Arc::try_unwrap(input).unwrap_or_else(|e| e.as_ref().clone());
+        let input = self.state().optimize(&input)?;
```

Comment:
That is important 👍 

I wonder if there is any test we can add to cover this case and prevent a regression?

Reply:
I have tried to add test to cover, but failed.
Because when I try to add test `explain`, but if sql is `explain`, df don't execute these code.